### PR TITLE
Arseni.chirkov/exercise04 locks

### PR DESCRIPTION
**Pull request by AChFr**
implemented post /credit endpoint and some locks. last bulletpoint of the forth issue commented in the  sendMessage controller

### CODE REVIEW

File: src/models/credit.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import mongoose from "mongoose";
+
+import database from "../database.js";
+
+const creditSchema = new mongoose.Schema({
+
+    amount: Number
+
+});
```
Comment (Teo-Lopez) [inline src/models/credit.js:9]: This schema is too basic, it should include the status for the transaction (OK, ERROR, TIMEOUT)

File: src/clients/topUpCredit.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import Credit from "../models/credit.js";
+
+export default async (creditAmount) => {
+
+  try {
+    const existingCredit = await Credit.findOne()
```
Comment (Teo-Lopez) [inline src/clients/topUpCredit.js:11]: This doesn't make much sense without some type of query. I will always return the first element in the collection (or a random one, not sure on the behaviour right now).

Also if you're going to find and then update you can use findAndUpdate

File: src/clients/topUpCredit.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+      const startingCredit = new Credit({ amount: 50 })
+      const newCredit = await startingCredit.save()
+
+      console.log("credito abierto")
+      return newCredit;
+    }
+
+
+
+  } catch (err) {
```
Comment (Teo-Lopez) [inline src/clients/topUpCredit.js:30]: Be very careful with your code style, white spaces, line breaks etc, there are lots of inconsistency

File: src/clients/checkCredit.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+        if ((existingCredit) && (existingCredit.amount - messageCost >= 0)) {
+            return true
+        } else {
+            return false
+        }
+    } catch (err) {
+        console.log("Error while checking balance", err);
+    }
+}
```
Comment (Teo-Lopez) [inline src/clients/checkCredit.js:16]: You are naming this function as checkCredit when it cheks and updates the credit. It's even more confusing on other files when you call it 

If I read this code I might try to call this functions multiple times to check if I have credit, each time I would be losing money 😱

**Review by Teo-Lopez — COMMENTED** [review]
Keep up the good work and good luck with those replicas

File: src/models/credit.js (hunk 1 above)
Comment (AChFr) [reply src/models/credit.js:9]: the statuses are  for the message Model. there is only one Entry in the credit database, and it has no status, its shows only the current balance.

**Pull request merged**